### ISSUE TITLE
Trouble compiling on Ubuntu

### ISSUE THREAD

**Issue by kimimaru4000** [fuzziqersoftware/gctools#1]
I'm running Ubuntu 17.10 in a VM, and I'm having issues compiling gctools. I followed all the instructions and have successfully installed all dependencies beforehand.
```
afsdump.cc In function `int main(int, char**)`:
afsdump.cc:40:24: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
    for (size_t x = 0; x < header->num_files; x)) {
```

If I fix this error, I end up with:
```
afsdump.cc:4:10: fatal error: /usr/local/include/phosg/Encoding.hh: Permission denied
 #include <phosg/Encoding.hh>
```
I ran `make` both with and without sudo and experience the same issues. Is there a solution to this?

**Comment by kimimaru4000** [comment]
After using `chmod` to set permissions on the .cc files and disabling `Werror` in the makefile, I ran into this issue:
```
In function `maybe_add_atfork_handler()`: /home/phosgdir/Process.cc:201: undefined reference to `pthread_atfork`
```
I'm unsure what's going on. I have the latest version of each dependency installed - and both Python 2.7 and Python 3+; do I need a specific version of each? I'm interested in building just afsdump, and this happens even if I attempt to compile it separately.

**Comment by fuzziqersoftware** [comment]
Try adding `-lpthread` to CXXFLAGS. I haven't built gctools on Ubuntu before, but I'll try it out sometime soon (hopefully today) and fix all of these issues - will update here when this is done.

**Comment by fuzziqersoftware** [comment]
da383109ac should fix this. Feel free to reopen this issue if it doesn't.

**Comment by kimimaru4000** [comment]
Thank you for getting on this so quickly! It got through more of it this time, but I'm getting an error that it can't find **phosg-audio/File.hh**. I don't see that file in my phosg folder nor in the phosg repo. Here's the full compiler message:
`aaf.cc:10:10: fatal error: phosg-audio/File.hh: No such file or directory
 #include <phosg-audio/File.hh>`

Unfortunately, I don't have the ability to reopen the issue.

**Comment by fuzziqersoftware** [comment]
Ah, you'll also have to install phosg-audio (https://github.com/fuzziqersoftware/phosg-audio) if you want to build the binaries in the sms directory. I split phosg-audio out from phosg because a lot of my other projects don't need the OpenAL dependency; will update the README in this project appropriately.

**Comment by kimimaru4000** [comment]
Unfortunately, I'm having issues compiling phosg-audio now. It seems to depend on OpenAL:
`Constants.hh:7:10: fatal error: AL/al.h: No such file or directory
 #include <AL/al.h>`

**Comment by fuzziqersoftware** [comment]
phosg-audio depends on OpenAL, which should be easy to install (`sudo apt-get install libopenal-dev` should do it).

**Comment by kimimaru4000** [comment]
Ah that makes sense, I didn't have the dev tools for OpenAL installed! It got much further this time, but unfortunately it still failed. Here's the output:
```
smssynth.cc: In member function `std::vector<float> Renderer::render_time_step(size_t, size_t)`:
smssynth.cc:1150:17: error: `%-7g` directive output may be truncated writing between 7 and 13 bytes into a region of size 8 [-Werror=format-truncation=]
    vector<float> render_time_step(size_t queued_buffer_count = 0, size_t buffer_count = 0) {
smssynth.cc:1150:17: note: assuming directive output of 12 bytes
smssynth.cc:1278:15: note: `snprintf` output between 8 and 14 bytes into a destination of size 8
        snprintf(when_str, 8, "%-7g", when);
```
Thanks again for being so attentive!

**Comment by fuzziqersoftware** [comment]
This is probably fixed by 4fdd47104b. Thanks for being patient as we remote-debug all of this :)

**Comment by kimimaru4000** [comment]
It worked great and everything compiled just fine! Thank you so much!
